pkg/webui/console/views/device-add/wizard/join-settings-form/validation-schema.js

```diff
@@ -52,25 +52,31 @@ const validationSchema = Yup.object()
         })
       },
     ),
-    net_id: Yup.nullableString().emptyOrLength(
-      3 * 2,
-      Yup.passValues(sharedMessages.validateLength),
-    ), // 3 Byte hex.
-    resets_join_nonces: Yup.boolean().when('$lorawanVersion', {
-      // Verify if lorawan version is 1.1.0 or higher.
-      is: version => parseLorawanMacVersion(version) >= 110,
+    net_id: Yup.nullableString().when(['$includeAdvanced'], {
```

Comment:
I think any collapsible section should not make a difference for the validation schema since:

1.  The sections only control the visibility of certain fields and they should not have any other side effects for the fields (e.g. a filled input that got hidden should still be respected when submitting the field)
2.  There will never be required fields (without default value) inside hideable sections

This also means that when a hidden field is marked as invalid on submit, the section will automatically be expanded so that the validation error becomes visible.

In fact, I believe it will be easiest for us if instead of removing the fields of a section from the DOM altogether (which has side effects for formik), we should use a visibility styling instead (e.g. `display: none` or `visibility: hidden`).